### PR TITLE
Add Pods cache dir to ios/.gitignore

### LINKED ISSUE #9085: Significant project diffs after first iOS flutter build
The first iOS build after `flutter create` results in fairly massive diffs in the Xcode project files, in particular to the CocoaPods integration. There are also a lot of CocoaPods-related files where I'm not clear on whether these should be under source control or not.

1. Can we update the templates to reduce the pod-related diffs on first build?
1. Does the `ios/.gitignore` (or other `.gitignore`s) need to be updated?

## Steps to Reproduce

On a macOS host:
1. `flutter create foobar`
1. `cd foobar`
1. `git init; git add .; git commit -m 'Initial commit'`
1. Bring up an iPhone simulator
1. `flutter run -d iPhone`
1. `git diff`

### PR DESCRIPTION
**Pull request by mit-mit**
Fixes https://github.com/flutter/flutter/issues/9085

### CODE REVIEW

File: examples/flutter_gallery/ios/.gitignore [hunk 1]
```diff
@@ -36,3 +36,5 @@ Icon?
 /Flutter/Flutter.framework
 /Flutter/Generated.xcconfig
 /ServiceDefinitions.json
+
+Pods/
```
Comment (cbracken) [inline examples/flutter_gallery/ios/.gitignore]: nit: blank line, missing newline at end

**Review by cbracken — APPROVED** [review]
LGTM. Thanks. 

On my phone so wasn't able to expand the diff and see if we're using blank lines as section dividers.

**Review by jakobr-google — APPROVED** [review]
LGTM

**Comment by mit-mit** [comment]
Merging. The broken flutter-build appears to be a flake as it passed in a re-run.

File: examples/flutter_gallery/ios/.gitignore (hunk 1 above)
Comment (mit-mit) [inline examples/flutter_gallery/ios/.gitignore]: Added the missing newline.

The blank line was intentional; the file is already formatted into logical groupings, and I wanted to stay consistent with that.

Comment (cbracken) [inline examples/flutter_gallery/ios/.gitignore]: Thanks! Wondered if that might be the case -- github doesn't allow expanding collapsed regions on phones :(

(and thanks for making this change!)

**Pull request merged**